src/PrestaShopBundle/Tests/Localization/CLDR/DataReaderTest.php

```diff
@@ -0,0 +1,472 @@
+<?php
+/**
+ * 2007-2017 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Tests\Localization;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShopBundle\Localization\CLDR\DataReader;
+
+class DataReaderTest extends TestCase
+{
+    /**
+     * @var DataReader
+     */
+    protected $reader;
+
+    public function setUp()
+    {
+        $this->reader = new DataReader();
+    }
+
+    /**
+     * Given a valid locale code
+     * When reading this locale data
+     * Retrieved data should match the expectation
+     *
+     * @param $localeCode
+     * @param $expectedData
+     *
+     * @dataProvider provideValidLocaleData
+     */
+    public function testItReadsLocaleData($localeCode, $expectedData)
+    {
+        $localeData = $this->reader->getLocaleByCode($localeCode);
+
+        foreach ($expectedData as $property => $value) {
+            // Assertions on number symbols lists (array of NumberSymbolList objects)
+            if ($property === 'numberSymbols') {
+                $expectedSymbolsSets = $value;
+                foreach ($expectedSymbolsSets as $expectedNumberingSystem => $expectedSymbolsSet) {
+                    foreach ($expectedSymbolsSet as $symbolName => $symbolValue) {
+                        $this->assertSame(
+                            $symbolValue,
+                            $localeData->numberSymbols[$expectedNumberingSystem]->$symbolName,
+                            'For "' . $localeCode . '" locale, "' . $expectedNumberingSystem . '" numbering system, "'
```

Comment:
Just FYI you can use variable interpolation, I think it's more readable than concatenation.